test/functional/connections_survive_primary_step_down_spec_tests.js

```diff
@@ -0,0 +1,299 @@
+'use strict';
+
+const chai = require('chai');
+const expect = chai.expect;
+
+let client;
+describe('Connections survive primary step down', function() {
+  beforeEach(function() {
+    client = this.configuration.newClient(
+      { w: 1 },
+      { poolSize: 1, retryWrites: false, useUnifiedTopology: true }
+    );
+  });
+
+  afterEach(function() {
+    return client.close();
```

Comment:
It looks like in a bunch of the tests you are calling `client.close`. Can we get rid of those and rely on the `afterEach`?